pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +91,507 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
+				}
+				if expectedFood < 0 {
+					expectedFood = 0
+				}
+
 			}
-			a.foodEaten = foodEaten
-			if a.foodEaten > 0 {
-				a.daysHungry = 0
+
+			// ------------------ Run() Block A.2: Adjusts mood w.r.t. change in floor and the expected food ------------------
+
+			// Update Behaviour: Higher in the tower (lower floor number) increases greed/decreases kindess and vice versa
+			if currentFloor < a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1] {
+				// A higher openess will more significantly reduce greedines and increase kindness
+				a.behaviour.greediness -= int(float64(a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1]/currentFloor) * (float64(a.personality.openness) / 50))
+				a.behaviour.kindness += int(float64(a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1]/currentFloor) * (float64(a.personality.openness) / 50))
+			} else {
+				// A lower openess will more significantly increase greedines and reduce kindness
+				a.behaviour.greediness += int(float64(currentFloor/a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1]) * (float64(100-a.personality.openness) / 50))
+				a.behaviour.kindness -= int(float64(currentFloor/a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1]) * (float64(100-a.personality.openness) / 50))
+			}
+
+			// Greediness is affected by the amount of food expected on this floor as estimated from past experience
+			a.behaviour.greediness -= int(float64(a.opMem.currentFloorRisk) * (float64(a.personality.conscientiousness) / 50))
+			foodReqCtoW := int(health.FoodRequired(healthInfo.HPCritical, healthInfo.WeakLevel, healthInfo))
+			if expectedFood < foodReqCtoW { //greediness only begins to be effected once the expected food is below a crticial threshold
+				a.opMem.currentFloorRisk = (foodReqCtoW - expectedFood)
+				a.behaviour.greediness += int(float64(a.opMem.currentFloorRisk) * (float64(a.personality.conscientiousness) / 50))
 			} else {
-				a.daysHungry++
+				a.opMem.currentFloorRisk = 0
 			}
-			a.Log("Team 7 has seen the platform:", infra.Fields{"foodEaten": a.foodEaten, "health": a.HP(), "daysHungry": a.daysHungry})
-			a.seenPlatform = true
+			a.TreatyOnFloorChange()
+		}
+		a.opMem.orderPrevFloors = append(a.opMem.orderPrevFloors, currentFloor) //append current floor to floor tracker
+	} else { //increment currentDayonFloor counter
+		a.opMem.currentDayonFloor++
+	}
+}
+
+func (a *CustomAgent7) manageMessages() {
+	UserID := a.ID()
+	currentFloor := a.Floor()
+	// ------------------ Run() Block B: Receive messages ------------------
+
+	receivedMsg := a.ReceiveMessage()
+	if receivedMsg != nil {
+		receivedMsg.Visit(a)
+	} else {
+		a.Log("No Messages")
+	}
+
+	// Messaging
+	if a.opMem.prevAge < a.Age() {
+		a.opMem.prevAge = a.Age()
+		a.opMem.msg1Sent = false
+		a.opMem.msg2Sent = false
+		a.opMem.msg3Sent = false
+	}
+
+	if !a.opMem.msg1Sent {
+		msg := messages.NewAskHPMessage(UserID, currentFloor, currentFloor+1)
```

Comment:
Can these 2 happen in the same tick? aka: two messages are sent in the same tick?